### PR TITLE
Tko jan24

### PR DESCRIPTION
**Pull request by tkoukkari**
* Feature: Adds new table keywords for handling column headers:
  * GetColHeaderCount
  * GetColHeader
  * VerifyColHeader
* Feature: Support Robot Framework 7 data model in TypeSecret
* Fix: Handle booleans given in string format in OpenBrowser prefs (i.e… "True")
* Dependencies: 
  * Allow Robot Framework 7
  * Bumped lowest allowed Pillow version 
* Updated workflow actions to newer versions

### CODE REVIEW

File: QWeb/keywords/cookies.py [hunk 1]
```diff
@@ -64,7 +64,7 @@ def list_cookies() -> list[dict[str, Any]]:
     \`DeleteAllCookies\`, \`IsCookie\`
     """
     cookies_list = cookies.get_cookies()
-    logger.info(cookies_list)
+    logger.info(f"{cookies_list}")
```
Comment (turunenm) [inline QWeb/keywords/cookies.py:67]: This is off topic from the PR but I've been wondering what is the rationale to info logging the cookies automatically.
I have some scripts where I'm looking for cookie values to update on a rapid pace and it clogs up the report.

Maybe add argument `log_cookies=True` to provide control over it for the user?

File: QWeb/keywords/table.py [hunk 2]
```diff
@@ -295,3 +296,199 @@ def get_table_row(
...
+    \`GetColumn\`, \`VerifyColumn\`, \`UseTable\`, \`VerifyTable\`
+    """
+    if not isinstance(ACTIVE_TABLE, Table):
+        raise QWebInstanceDoesNotExistError('Table has not been defined with UseTable keyword')
+    table = Table.ACTIVE_TABLE.update_table()
+    return len(table.get_columns())
+
+
+@keyword(tags=("Tables", "Getters"))
+def get_column(index: Optional[int] = None) -> Union[str, List[str]]:
```
Comment (turunenm) [inline QWeb/keywords/table.py]: I would change this to `get_header_column` or `get_table_header_column`

Compare the behavior to `get_table_row` which returns the first row which contains the sought text.
This kw only looks for the first row and it's column count which differs from the existing row kw

File: QWeb/keywords/table.py [hunk 3]
```diff
@@ -295,3 +296,199 @@ def get_table_row(
     if isinstance(ACTIVE_TABLE, Table) is False:
         raise QWebInstanceDoesNotExistError('Table has not been defined with UseTable keyword')
     return table.get_row(locator, anchor, row_index=True, **kwargs)
+
+
+@keyword(tags=("Tables", "Getters"))
+def get_column_count() -> int:
```
Comment (turunenm) [inline QWeb/keywords/table.py]: I would change this to `get_header_column_count` or `get_table_header_column_count`

Rows can have differing amounts of columns which are close impossible to account for in a sane manner.
But even so the kw name should describe as accurately as possible what it actually does.

File: QWeb/keywords/table.py [hunk 4]
```diff
@@ -295,3 +296,199 @@ def get_table_row(
...
+                  **kwargs) -> bool:
+    r"""Verifies that a column with given index matches the expected text.
+
+    Note: only visible columns may be counted in some dynamic tables.
+
+    Note2: If column header has attribute 'aria-label', it's value is considered as expected value.
+    If not and 'title' attirbute exists, then title is considered as expected value.
+    If neither aria-label or title exists, then column text is considered as expected value.
+    This is done due to fact that in many web applications header element text contains other texts
+    than what is visible to end users.
```
Comment (turunenm) [inline QWeb/keywords/table.py]: Doc and behavior do not match.

Looking at `_get_column_header_texts` -> title > aria-label > text

File: QWeb/keywords/table.py [hunk 5]
```diff
@@ -295,3 +296,199 @@ def get_table_row(
...
+        raise QWebValueError('Column index should be a positive integer')
+
+    if len(columns) < index:
+        raise QWebValueError(f'Column index out of range: {index=}, {len(columns)=}')
+
+    return column_texts[index - 1]
+
+
+@keyword(tags=("Tables", "Verification"))
+def verify_column(expected: str,
```
Comment (turunenm) [inline QWeb/keywords/table.py]: I would change this to `verify_header_column`

The kw only looks for the first rows column texts and the name should reflect it

**Review by samijt — APPROVED** [review]

File: QWeb/internal/secrets.py [hunk 6]
```diff
@@ -101,6 +109,34 @@ def _filtered_start_keyword(keyword: Keyword) -> None:
             LOGGER._other_loggers[0].log_message = lambda x: None
 
 
+def _filtered_start_library_keyword(data: Keyword, implementation: Keyword, result: Keyword):
```
Comment (turunenm) [inline QWeb/internal/secrets.py:112]: Add docstring.
Specify that this is function for RFW version 7+

File: QWeb/internal/secrets.py [hunk 7]
```diff
@@ -130,6 +166,37 @@ def _filtered_end_keyword(keyword: Keyword) -> None:
             LOGGER._other_loggers[0].log_message = debugfile_log
 
 
+def _filtered_end_library_keyword(data: Keyword, implementation: Keyword, result: Keyword) -> None:
+    """Modify Robot FW internal function "end_keyword".
+
+    This function removes secret data from "end keyword"
+    logs and returns previous log level.
+    """
```
Comment (turunenm) [inline QWeb/internal/secrets.py]: Specify that this function is for RFW version 7+

**Review by irosengr — APPROVED** [review]

**Review by turunenm — COMMENTED** [review]
IMO column kw names should be changed to reflect their actual behavior to avoid confusion.

File: QWeb/keywords/cookies.py (hunk 1 above)
Comment (tkoukkari) [reply QWeb/keywords/cookies.py:67]: Good suggestion, will be handled in own pr / part of one release after this one.

File: QWeb/keywords/table.py (hunk 3 above)
Comment (tkoukkari) [reply QWeb/keywords/table.py]: Renamed in latest commit

File: QWeb/keywords/table.py (hunk 2 above)
Comment (tkoukkari) [reply QWeb/keywords/table.py]: Renamed as:
- GetColumnCount -> GetColHeaderCount
- GetColumn -> GetColHeader
- VerifyColumn -> VerifyColHeader

File: QWeb/keywords/table.py (hunk 3 above)
Comment (tkoukkari) [reply QWeb/keywords/table.py]: Renamed:
- GetColumnCount -> GetColHeaderCount
- GetColumn -> GetColHeader
- VerifyColumn -> VerifyColHeader

File: QWeb/keywords/table.py (hunk 4 above)
Comment (tkoukkari) [reply QWeb/keywords/table.py]: Thanks, did not notice this. Fixed in latest commit and also the typo in help text fixed

File: QWeb/internal/secrets.py (hunk 6 above)
Comment (tkoukkari) [reply QWeb/internal/secrets.py:112]: added and also end_library_keyword docstring modified

File: QWeb/internal/secrets.py (hunk 7 above)
Comment (tkoukkari) [reply QWeb/internal/secrets.py]: done

File: QWeb/keywords/table.py (hunk 5 above)
Comment (tkoukkari) [reply QWeb/keywords/table.py]: renamed

**Review by turunenm — APPROVED** [review]
lgtm

**Pull request merged**